### PR TITLE
Fix/load search results only once

### PR DESCRIPTION
**Pull request by ulferts**
Reduces the times, search results are loaded from three to one and by that improves performance quite a bit.

https://community.openproject.com/projects/openproject/work_packages/29715

### CODE REVIEW

**Comment by ulferts** [comment]
The tests broken are unit tests which I cannot have caused as I only changed front end code.

File: frontend/src/app/modules/global_search/global-search-work-packages.component.ts [hunk 1]
```diff
@@ -108,13 +100,13 @@ export class GlobalSearchWorkPackagesComponent extends WorkPackageEmbeddedTableC
...
     )
     .pipe(
+      skip(1),
       distinctUntilChanged(),
       debounceTime(10),
       untilComponentDestroyed(this)
     )
     .subscribe(([newSearchTerm, newProjectScope]) => {
       this.WpFilter.visible = false;
-      this.firstTimeLoadForm = true;
```
Comment (wielinde) [inline frontend/src/app/modules/global_search/global-search-work-packages.component.ts:117]: @ulferts If we remove this line, we run into trouble. Say, you search, open "Advanced Filters" then the searched term shall be set as value of the SearchFilter. Then you trigger a new search through the GlobalSearchInput and open "Advanced filters" again. If we remove this line, the value of the SearchFilter does not update, and that is bad. Obviously, loading the form everytime we perform a new search isn't optimal either but at least "it works".

**Review by wielinde — CHANGES_REQUESTED** [review]
@ulferts please see my comment/discussion above.

File: frontend/src/app/modules/global_search/global-search-work-packages.component.ts (hunk 1 above)
Comment (ulferts) [reply frontend/src/app/modules/global_search/global-search-work-packages.component.ts:117]: Will take a look at it, @wielinde. Thanks for finding.

**Comment by ulferts** [comment]
Pushed a new version where I removed the `firstTimeLoadForm` variable altogether. As  `loadQuery` is now only called in cases where the form is also required to be loaded, removing should be save.

**Comment by wielinde** [comment]
Hey @ulferts, on a first check I found that the search filter value in the advanced filters does not updated when you fire a new search from within the global search input.

**Comment by ulferts** [comment]
@wielinde will take a look

**Comment by wielinde** [comment]
@ulferts thank you!

**Pull request merged**